src/test/java/org/javarosa/core/model/instance/TreeReferenceSerializationTest.java

```diff
@@ -0,0 +1,92 @@
+package org.javarosa.core.model.instance;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.javarosa.core.test.AnswerDataMatchers.stringAnswer;
+import static org.javarosa.core.util.BindBuilderXFormsElement.bind;
+import static org.javarosa.core.util.XFormsElement.body;
+import static org.javarosa.core.util.XFormsElement.head;
+import static org.javarosa.core.util.XFormsElement.html;
+import static org.javarosa.core.util.XFormsElement.input;
+import static org.javarosa.core.util.XFormsElement.mainInstance;
+import static org.javarosa.core.util.XFormsElement.model;
+import static org.javarosa.core.util.XFormsElement.t;
+import static org.javarosa.core.util.XFormsElement.title;
+
+import java.io.IOException;
+import org.javarosa.core.test.Scenario;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.junit.Test;
+
+public class TreeReferenceSerializationTest {
+    @Test
+    public void mainInstanceName_afterDeserialization_isNull() throws IOException, DeserializationException {
```

Comment:
`..isStillNull`

Reply:
Maybe split this up or make it clearer this is testing three things

Reply:
Thank you, that does make things a lot clearer!